webapp/src/hooks/useConfirm.ts

```diff
@@ -58,45 +58,53 @@ function errorFunction() {
  * - `yes`: A function that resolves the promise with `true`.
  * - `no`: A function that resolves the promise with `false`.
  * - `cancel`: A function that resolves the promise with `null`.
+ * - `data`: The data passed to the `showConfirm` function.
  */
-function useConfirm() {
+function useConfirm<TData = undefined>() {
   const [isPending, setIsPending] = useState(false);
+  const [data, setData] = useState<TData>();
   const isPendingRef = useUpdatedRef(isPending);
   const yesRef = useRef<VoidFunction>(errorFunction);
   const noRef = useRef<VoidFunction>(errorFunction);
   const cancelRef = useRef<VoidFunction>(errorFunction);
 
-  const showConfirm = useCallback(() => {
-    if (isPendingRef.current) {
-      throw new Error("A promise is already pending");
-    }
+  const showConfirm = useCallback(
+    (params: TData extends undefined ? void : { data: TData }) => {
```

Comment:
you're using `void` as a type parameter in a conditional type.
in this case (value types) it's preferable to use `undefined` instead of `void`

optional: consider defining the type outside of the function for better readability
```ts
  type ShowConfirmParams = TData extends undefined ? undefined : { data: TData };
```

Reply:
`void` allows to define the field as optional (doesn't work with `undefined`)

Reply:
Ok i understand, but `void` typically represents the absence of a return value, not the absence of an argument
you should handle this case explicitly